---
title: 异常处理
author: peppacat
date: 2023-03-01
category: Jekyll
layout: post
---

# 介绍
 go中的异常处理 defer panic recover
 抛出一个panic异常,在defer中通过recover捕获处理

```go
package main

import "fmt"

func main() {

	// go中的异常处理 defer panic recover
	// 抛出一个panic异常,在defer中通过recover捕获处理

	test()
	fmt.Println("Hello world")

}

func test() {
	// 匿名函数
	defer func() {
		error := recover()
		if error != nil {
			fmt.Println("error = ", error)

		}
	}()
	var n1 int = 10
	var n2 int = 0
	var n3 int = n1 / n2
	fmt.Println("res = ", n3)
}


```